pkgs/by-name/ca/cargo-make/package.nix

```diff
@@ -37,16 +39,31 @@ rustPlatform.buildRustPackage rec {
       darwin.apple_sdk.frameworks.SystemConfiguration
     ];
 
-  postInstall = ''
-    installShellCompletion extra/shell/*.bash
-  '';
+  postInstall =
+    ''
+      installShellCompletion extra/shell/*.bash
+    ''
+    + lib.optionalString (stdenv.buildPlatform.canExecute stdenv.hostPlatform) ''
+      export HOME=$(mktemp -d)
+      $out/bin/cargo-make make --completion zsh
+      installShellCompletion --cmd cargo-make --zsh "$HOME/.zfunc/_cargo-make"
+    '';
```

Comment:
I'll omit this change because I don't have confidence around that. https://github.com/sagiegurari/cargo-make/pull/1218#issuecomment-2599822933